R/Utilities.R

```diff
@@ -64,15 +64,25 @@ autoclean <- function(.data) {
     toupper(.data$DetectionQuantitationLimitMeasure.MeasureUnitCode)
   # .data$BiologicalIntentName = toupper(.data$BiologicalIntentName)
   
-  # Remove duplicate rows
-  .data <- .data[!duplicated(.data), ]
+  # Remove duplicate rows - turned into a test because duplicated() takes a long
+  # time acting on all columns in a large dataset.
+  if(!length(unique(.data$ResultIdentifier))==dim(.data)[1]){
+    print("Duplicate records may be present. Filtering to unique records. This may take a while on large datasets.")
```

Comment:
change to "Duplicate records are present" ?

Reply:
I wrote "may be present" because I could potentially see someone using autoclean on a dataset where they joined detection limit data to result data and thus have unique rows with the same result identifier and different detection limits. In this case, the function will check to make sure these cases are truly unique.